### DISCUSSION TITLE
how to export class's module?

### DISCUSSION THREAD

**Discussion by shemeshg** [hsutter/cppfront#1059] [Q&A] [answered]
It seems that `cppfront -p`
```
module;

export module mymodule;
```

Make cpp2 complain it is not pure cpp2, 
`-im` seems to overcome this, but then it ignore export of a class

```
export mytype: type = {
    data: i32;   // some data member (private by default)

    print: (this, msg: std::string) = {
        std::cout << data << msg;
                 // "data" is shorthand for "this.data"
    }

    // ...
}
```
export here considered cpp1 and ignored.

**Comment by zaucy** [comment, marked as answer]
Authoring modules aren't supported yet, I believe

* https://github.com/hsutter/cppfront/issues/269
* https://github.com/hsutter/cppfront/pull/569#issuecomment-1674169682

I would expect what you have above would work once supported.

**Comment by hsutter** [comment]
Thanks! Yes, module export is not supported yet -- I'm waiting for the major compilers to have stable support for modules.
